indico/modules/users/controllers.py

```diff
@@ -382,3 +431,45 @@ def _process(self):
         target = User.get_one(request.args['target'])
         errors, warnings = _get_merge_problems(source, target)
         return jsonify(errors=errors, warnings=warnings)
+
+
+class RHRegistrationRequestList(RHAdminBase):
+    """List all registration requests"""
+
+    def _process(self):
+        requests = RegistrationRequest.query.order_by(RegistrationRequest.email).all()
+        return WPUsersAdmin.render_template('registration_requests.html', pending_requests=requests)
+
+
+class RHRegistrationRequestBase(RHAdminBase):
+    """Base class to process a registration request"""
+
+    CSRF_ENABLED = True
+
+    def _checkParams(self, params):
+        RHAdminBase._checkParams(self, params)
+        self.request = RegistrationRequest.get_one(request.view_args['request_id'])
+
+
+class RHAcceptRegistrationRequest(RHRegistrationRequestBase):
+    """Accept a registration request"""
+
+    def _process(self):
+        user, identity = register_user(self.request.email, self.request.extra_emails, self.request.user_data,
+                                       self.request.identity_data, self.request.settings)
+        tpl = get_template_module('users/emails/registration_request_accepted.txt', user=user)
+        send_email(make_email(self.request.email, template=tpl))
+        db.session.delete(self.request)
```

Comment:
What happens if we `register_user` but `self.request` is not deleted? I'm asking in case we call `request_user` from `indico shell` wondering if deletion should be part of the operation or if having dangling requests for users that already exist is acceptable.